tasks/debugging/dump.py

```diff
@@ -0,0 +1,447 @@
+            print("No symbols found")
+            return
+        print("Symbols:")
+        for symbol_file in find_symbol_files_for_platform(ca.target_platform, syms):
+            print('\t', Path(symbol_file).resolve())
+
+
+@task(
+    help={
+        "job_id": "Any job ID from the pipeline to download the symbols from",
+        "version": "The released Agent version to download symbols for (e.g. 7.57.0)",
+        "platform": f"The target platform ({', '.join(PLATFORM_CHOICES)})",
+        "arch": f"The target architecture ({', '.join(ARCH_CHOICES)})",
+    }
+)
+def get_debug_symbols(
+    ctx, job_id: str | None = None, version: str | None = None, platform: str | None = None, arch: str | None = None
+):
+    """
+    Download debug symbols for a specific Agent version or pipeline
+    """
+    ca = get_crash_analyzer(platform=platform, arch=arch)
+
+    if version:
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            get_debug_symbols_for_version(version, ca.target_platform, ca.target_arch, tmp_dir)
+            syms = ca.symbol_store.add(version, ca.target_platform, ca.target_arch, tmp_dir)
+    elif job_id:
+        ca.select_project(get_gitlab_repo())
+        version, syms = get_symbols_for_job_id(ca, job_id)
+
+    print(f"Symbols for {version} in {syms}")
+
+
+def add_gitlab_job_artifacts_to_artifact_store(
+    artifact_store: ArtifactStore, project: Project, job_id: str
+) -> Artifacts:
+    with tempfile.TemporaryDirectory() as temp_dir:
+        download_job_artifacts(project, job_id, temp_dir)
+        project_id = project.name
+        job_id = str(job_id)
+        return artifact_store.add(project_id, job_id, temp_dir)
+
+
+def get_symbols_for_job_id(ca: CrashAnalyzer, job_id: str) -> tuple[str | None, Path | None]:
+    if not ca.active_project:
+        raise ValueError("No active project set")
+    project_id = ca.active_project.name
+    # check if we already have the symbols for this job
+    artifact = ca.artifact_store.get(project_id, job_id)
+    syms = None
+    if artifact and artifact.version:
+        syms = ca.symbol_store.get(artifact.version, ca.target_platform, ca.target_arch)
+        if syms:
+            return artifact.version, syms
+    # Need to get the symbols from the package build job in the pipeline
+    package_job_id = get_package_job_id(ca.active_project, job_id, ca.target_platform, ca.target_arch)
+    if not package_job_id:
+        raise Exception(f"Could not find package job for job {job_id}")
+    package_artifacts = get_or_fetch_artifacts(ca.artifact_store, ca.active_project, package_job_id)
+    path = package_artifacts.get()
+    if not path:
+        raise FileNotFoundError(f"No artifacts found for job {package_job_id}")
+    archives = find_platform_debug_artifacts(ca.target_platform, path)
+    # TODO: hacky way to get the version from the archive name, is there a better way?
+    #       the setup-agent-version job stores the version in a private bucket.
```

Comment:
I think we should be able to compute using that function: https://github.com/DataDog/datadog-agent/blob/b7383882501637956482b60e1e112963664e74a8/tasks/agent.py#L777-L789
But note sure what parameters you would need to pass.
If that works like that I guess this is fine for a first PR

Reply:
this leads to `query_version` which runs git commands. If those commands support being given a refspec i might be able to give it commit hash from the pipeline, but it would only work if you have the refspec locally, too. it might be nice to have a clone of the repo at that commit, though, so the source matches.